app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,19 @@
+class CommentsController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+    new_comment = current_user.comments.new(params.require(:comment).permit(:text))
+    new_comment.post_id = @post.id
+    new_comment.update_comments_counter
+    if new_comment.save
+      redirect_to "/users/#{@post.user_id}/posts/#{@post.id}", notice: 'Success!'
+    else
+      render :new, alert: 'Error occured!'
+    end
+  end
+
+  # private
+
+  # def comment_params
+  #   params.require(:comment).permit(:text)[:text]
+  # end
```

Comment:
- [ ] I would recommend including the private method, it makes the application secure by permitting only limited attributes.

- [ ] Also add `new` method for CommentsController like PostsController. Also add `new` path for comments in `routes.rb`

```ruby
class CommentsController < ApplicationController
 def new
    @comment = Comment.new
  end
  ....
end
```

Reply:
Thank you for your suggestion. Fixed those errors.